java-cfenv-boot/src/main/java/io/pivotal/cfenv/spring/boot/GenAIChatCfEnvProcessor.java

```diff
@@ -15,37 +15,47 @@
  */
 package io.pivotal.cfenv.spring.boot;
 
+import java.util.ArrayList;
 import java.util.Map;
 
 import io.pivotal.cfenv.core.CfCredentials;
 import io.pivotal.cfenv.core.CfService;
 
 /**
  * Retrieve GenAI on Tanzu Platform properties from {@link CfCredentials} and
- * set {@literal spring.ai}
+ * set {@literal spring.ai.openai.chat}
  * Boot properties.
  *
  * @author Stuart Charlton
+ * @author Ed King
  **/
-public class GenAICfEnvProcessor implements CfEnvProcessor {
+public class GenAIChatCfEnvProcessor implements CfEnvProcessor {
 
 	@Override
 	public boolean accept(CfService service) {
-		return service.existsByTagIgnoreCase("genai") ||
-				service.existsByLabelStartsWith("genai");
+		boolean isGenAIService = service.existsByTagIgnoreCase("genai") || service.existsByLabelStartsWith("genai");
+		if (isGenAIService) {
+			ArrayList<String> modelCapabilities = (ArrayList<String>) service.getCredentials().getMap().get("model_capabilities");
+			return modelCapabilities.contains("chat");
+		} else {
+			return false;
```

Comment:
nitpick: I would use the interface type List instead of the concrete type ArrayList. My java is also rusty though, is there any specific reason that made you use ArrayList? I think at least you should be able to use List in the left hand side

Reply:
Thanks @pivotal-marcela-campo, I have removed the redundant "else" blocks. Re: usage of ArrayList, I simply copied from [some existing code](https://github.com/pivotal-cf/java-cfenv/blob/v3.1.5/java-cfenv-boot/src/main/java/io/pivotal/cfenv/spring/boot/CassandraCfEnvProcessor.java#L48) in the `CassandraCfEnvProcessor`.